core/src/main/java/org/pentaho/reporting/platform/plugin/ParameterDependencyGraph.java

```diff
@@ -233,7 +243,7 @@ private boolean processDependentParameters( MasterReport report,
     }
 
     final DataFactory dataFactoryForQuery = cdf.getDataFactoryForQuery( queryName );
-    if ( dataFactoryForQuery == null ) {
+    if ( dataFactoryForQuery == null || dataFactoryForQuery instanceof TableDataFactory ) {
```

Comment:
I'm sure this should be obvious, however, it looks surprising for an outsider. What is it with this type of data factory which makes us immediately return with an empty list? Please, consider adding a comment.

Reply:
Comment added.